cellxgene_resources/curation_qa.ipynb

```diff
@@ -656,6 +656,133 @@
     "    ][['donor_id','development_stage_ontology_term_id']]"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Check donor sex metadata against raw gene expression counts (genes found in PAR and NonPAR regions)**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Read in par regions\n",
+    "par_file = 'sex_analysis_regions/list_chrY_PAR_genes.txt'\n",
+    "nonpar_file =  'sex_analysis_regions/list_chrY_nonPAR_genes.txt'\n",
+    "par = pd.read_table(par_file, names=['par_genes'])\n",
+    "nonpar = pd.read_table(nonpar_file, names=['nonpar_genes'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Use raw matrix counts for sex analysis\n",
+    "raw_adata = ad.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)  # Create raw anndata if adata.raw exists otherwise raw_adata = adata\n",
+    "\n",
+    "#Extract regions from adata based on feature_name \n",
+    "par_adata = raw_adata[:,raw_adata.var.feature_name.isin(par['par_genes'])]  \n",
+    "nonpar_adata = raw_adata[:,raw_adata.var.feature_name.isin(nonpar['nonpar_genes'])] \n",
+    "region_dict = {'par': [par_adata], 'nonpar': [nonpar_adata]}\n",
+    "\n",
+    "#Check to see how many par/nonpar genes were found\n",
+    "print(f'% PAR genes found: {(par_adata.shape[1]/len(par[\"par_genes\"]))*100}')\n",
+    "print(f'% NonPAR genes found: {(nonpar_adata.shape[1]/len(nonpar[\"nonpar_genes\"]))*100}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Check for batch or library uuid fields for correct grouping\n",
+    "print(raw_adata.obs_keys)\n",
+    "library_id = 'batch'  # this variable will change depending on the dataset"
```

Comment:
We can't count on a "library" ID metadata field in all Datasets. We should try to avoid this customization.